### PR TITLE
feat: edit name in show app screen

### LINKED ISSUE #451: Rename existing connections/apps
Replacing connection/app is difficult when app
- is heavily integrated in other systems
- is isolated and you want to preserve balance

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #451

~⚠️ Needs design input on frontend (currently can break in small screens (<350px) and also doesn't have cancel option)~ Proceeding with this for now

<p>
Screenshot 2024-08-16 at 2 37 20 PM
Screenshot 2024-08-16 at 2 37 37 PM
</p>
<p>
Screenshot 2024-08-16 at 2 37 20 PM
Screenshot 2024-08-16 at 2 37 20 PM
</p>

### CODE REVIEW

File: frontend/src/screens/apps/NewApp.tsx [hunk 1]
```diff
@@ -180,6 +182,10 @@ const NewAppInternal = ({ capabilities }: NewAppInternalProps) => {
     }
 
     try {
+      if (apps?.some((existingApp) => existingApp.name === appName)) {
+        throw new Error("A connection with the same name already exists.");
```
Comment (rolznz) [inline frontend/src/screens/apps/NewApp.tsx:186]: this also happens if you click edit and then save without changing, and there is no cancel button

Comment (rolznz) [reply frontend/src/screens/apps/NewApp.tsx:186]: oops, wrong one.

File: frontend/src/screens/apps/ShowApp.tsx [hunk 2]
```diff
@@ -95,7 +103,15 @@ function AppInternal({ app, refetchApp, capabilities }: AppInternalProps) {
 
   const handleSave = async () => {
     try {
+      if (
+        isEditingName &&
+        apps?.some((existingApp) => existingApp.name === name)
+      ) {
+        throw new Error("A connection with the same name already exists.");
```
Comment (rolznz) [inline frontend/src/screens/apps/ShowApp.tsx:110]: this also happens if you click edit and then save without changing, and there is no cancel button

Comment (rolznz) [reply frontend/src/screens/apps/ShowApp.tsx:110]: Maybe just check if the app id is different too?

File: frontend/src/screens/apps/ShowApp.tsx [hunk 3]
```diff
@@ -111,10 +127,13 @@ function AppInternal({ app, refetchApp, capabilities }: AppInternalProps) {
       });
 
       await refetchApp();
-      setEditMode(false);
-      toast({ title: "Successfully updated permissions" });
+      setIsEditingName(false);
+      setIsEditingPermissions(false);
+      toast({
+        title: `Successfully updated ${isEditingName ? "name" : "permissions"}`,
```
Comment (rolznz) [inline frontend/src/screens/apps/ShowApp.tsx:133]: simply "Successfully updated connection" is probably enough

File: frontend/src/screens/apps/ShowApp.tsx (hunk 2 above)
Comment (im-adithya) [reply frontend/src/screens/apps/ShowApp.tsx:110]: Oh, lol true. Yup, will add app id check 👍

File: frontend/src/screens/apps/ShowApp.tsx [hunk 4]
```diff
@@ -105,7 +104,9 @@ function AppInternal({ app, refetchApp, capabilities }: AppInternalProps) {
     try {
       if (
         isEditingName &&
-        apps?.some((existingApp) => existingApp.name === name)
+        apps?.some(
+          (existingApp) => existingApp.name === name && existingApp.id != app.id
```
Comment (rolznz) [inline frontend/src/screens/apps/ShowApp.tsx:108]:
```suggestion
          (existingApp) => existingApp.name === name && existingApp.id !== app.id
```

**Review by rolznz — APPROVED** [review]

**Pull request merged**